### PR TITLE
remove-from-mesh: skip system namespace to remove sidecar

### PR DESCRIPTION
**Pull request by shamsher31**
To fix https://github.com/istio/istio/issues/28186

Before:
Where `local-path-provisioner` don't have sidecar as it is in system namespace, still we get.
```
$ istioctl experimental remove-from-mesh deploy local-path-provisioner -n local-path-storage
deployment "local-path-provisioner.local-path-storage" updated successfully with Istio sidecar un-injected.
```
```
$ istioctl experimental remove-from-mesh deployment istio-egressgateway  -n istio-system
Error: 1 error occurred:
	* failed to update deployment "istio-egressgateway.istio-system" for service "istio-egressgateway.istio-system" due to Deployment.apps "istio-egressgateway" is invalid: spec.template.spec.containers: Required value
```

After:
```
$ ./out/linux_amd64/istioctl x rm deploy local-path-provisioner -n local-path-storage
Error: namespace local-path-storage is a system namesapce and has no Istio sidecar injected
```
```
$ ./out/linux_amd64/istioctl x rm deploy istio-ingressgateway-5fc96f8db6-rh69b -n istio-system
Error: namespace istio-system is a system namesapce and has no Istio sidecar injected
```

### CODE REVIEW

File: istioctl/cmd/remove-from-mesh_test.go [hunk 1]
```diff
@@ -139,6 +139,20 @@ func TestRemoveFromMesh(t *testing.T) {
...
+			expectedException: true,
+			namespace:         "local-path-storage",
+			expectedOutput:    "Error: namespace local-path-storage is a system namesapce and has no Istio sidecar injected\n",
+		},
+		{
+			description:       "Invalid namespace - system namespace not allowed",
+			args:              strings.Split("experimental remove-from-mesh service istio-ingressgateway", " "),
+			expectedException: true,
+			namespace:         "istio-system",
+			expectedOutput:    "Error: namespace istio-system is a system namesapce and has no Istio sidecar injected\n",
```
Comment (shamsher31) [inline istioctl/cmd/remove-from-mesh_test.go:154]: Maybe a better message if the user tries to remove sidecar from the istio-system namespace?

File: istioctl/cmd/remove-from-mesh.go [hunk 2]
```diff
@@ -124,12 +130,14 @@ func svcUnMeshifyCmd() *cobra.Command {
 			if len(args) != 1 {
 				return fmt.Errorf("expecting service name")
 			}
+			ns := handlers.HandleNamespace(namespace, defaultNamespace)
+			if analyzer_util.IsSystemNamespace(resource.Namespace(ns)) || ns == constants.IstioSystemNamespace {
```
Comment (irisdingbj) [inline istioctl/cmd/remove-from-mesh.go:134]: Can you change the fixed `constants.IstioSystemNamespace`  to the value passed via `-i` to exclude istio's system namespace?

**Review by irisdingbj — COMMENTED** [review]
Thanks for the fix.  Please take a look at the handling of istio's system namespace.

File: istioctl/cmd/remove-from-mesh.go (hunk 2 above)
Comment (shamsher31) [reply istioctl/cmd/remove-from-mesh.go]: Done. Thanks.

**Review by irisdingbj — APPROVED** [review]

**Pull request merged**